### PR TITLE
View mark by parent bka

### PR DESCRIPTION
**Pull request by bodkia22**

### CODE REVIEW

File: src/app/components/children-info/children-info.component.html [hunk 1]
```diff
@@ -8,14 +8,49 @@
...
-    (click)="reset()" 
-    class="btn back-btn"  
-    *ngIf="user != null"
->
-    Back
+        (click)="reset()" 
+        class="btn back-btn"  
+        *ngIf="user != null"
+    >
+        Back
```
Comment (Karonuk) [inline src/app/components/children-info/children-info.component.html]: delete tabulation

**Review by Karonuk — APPROVED** [review]

**Review by Karonuk — DISMISSED** [review]

File: src/app/components/add-schedule-form/add-schedule-form.component.ts [hunk 2]
```diff
@@ -119,7 +122,7 @@ export class AddScheduleFormComponent implements OnInit {
       let request = {
         ...this.form.value,
         weekCopyCount: parseInt(this.form.value.weekCopyCount, 10),
-        lessonDate: this.form.value.lessonDate.toDateString()
+        lessonDate: this.form.value.lessonDate
```
Comment (DmMov) [inline src/app/components/add-schedule-form/add-schedule-form.component.ts]: What is this line for?

File: src/app/components/children-info/children-info.component.html [hunk 3]
```diff
@@ -14,8 +14,43 @@
...
+        (click)="changeTab('marks')"
+        [ngClass]="{active: activeTab == 'marks'}"
+    >
+        Marks
+    </button>
+</div>
 <yps-profile-card  
     [user]="user"
-    *ngIf="user != null"
+    *ngIf="user != null && activeTab=='profile'"
```
Comment (DmMov) [inline src/app/components/children-info/children-info.component.html]: Align!!!

File: src/app/components/children-info/children-info.component.ts [hunk 4]
```diff
@@ -14,22 +18,37 @@ export class ChildrenInfoComponent implements OnInit {
...
-    private httpOptionsService: HttpOptionsService
+    private httpOptionsService: HttpOptionsService,
+    private formBuilder: FormBuilder
   ) { }
 
   pupils: IUser[];
 
   user: IUser[] = null;
+  
+  selectedPupilId: number;
```
Comment (DmMov) [inline src/app/components/children-info/children-info.component.ts:27]: I think it is useless

File: src/app/components/children-info/children-info.component.ts [hunk 5]
```diff
@@ -14,22 +18,37 @@ export class ChildrenInfoComponent implements OnInit {
...
+  activeTab: string = "profile";
+
+  journalColumns: IJournalColumn[];
+  disciplines: IDisciplineToSelect[];
+  form: FormGroup;
+
+
+  changeTab(name: string){
+    this.activeTab = name;
+  }
```
Comment (DmMov) [inline src/app/components/children-info/children-info.component.ts]: Use arrow functions and use spaces where they are needed.

File: src/app/components/children-info/children-info.component.ts [hunk 6]
```diff
@@ -14,22 +18,37 @@ export class ChildrenInfoComponent implements OnInit {
...
+  
+  selectedPupilId: number;
+  activeTab: string = "profile";
+
+  journalColumns: IJournalColumn[];
+  disciplines: IDisciplineToSelect[];
+  form: FormGroup;
+
+
+  changeTab(name: string){
```
Comment (DmMov) [inline src/app/components/children-info/children-info.component.ts]: There is an extra line here.

File: src/app/components/children-info/children-info.component.ts [hunk 7]
```diff
@@ -14,22 +18,37 @@ export class ChildrenInfoComponent implements OnInit {
...
+    this.form.reset();
   }
 
   selectPupil = (id: number) => {
     this.http.get(apiUrl + "/Parents/GetPupilAsParent/" + id, this.httpOptionsService.options)
     .subscribe(
       (successRes: IUser[]) => {
-        console.log(this.user = successRes);
+        this.user = successRes;
+        this.selectedPupilId = id;
```
Comment (DmMov) [inline src/app/components/children-info/children-info.component.ts:47]: Very interesting.

Comment (DmMov) [reply src/app/components/children-info/children-info.component.ts:47]: And funny.

File: src/app/pages/pupil-journal/pupil-journal.component.html [hunk 8]
```diff
@@ -1,13 +1,13 @@
...
- <span class="no-marks" *ngIf="journalColumns?.length == 0">
-    you don't have any marks yet.
-  </span>
- <yps-pupil-journal-column-list [journalColumns]= "journalColumns"></yps-pupil-journal-column-list>
\ No newline at end of file
+</form>
+<span class="no-marks" *ngIf="journalColumns?.length == 0">
+  you don't have any marks yet.
+</span>
+<yps-pupil-journal-column-list [journalColumns]="journalColumns"></yps-pupil-journal-column-list>
```
Comment (DmMov) [inline src/app/pages/pupil-journal/pupil-journal.component.html]: If you are trying to align it, align it all and do it right.

File: src/app/pages/pupil-journal/pupil-journal.component.ts [hunk 9]
```diff
@@ -28,7 +28,7 @@ export class PupilJournalComponent implements OnInit {
       "disciplineId": [null, requiredValidator("disciplines is required")]
     })
 
-    this.http.get(`${apiUrl}/Disciplines/GetAllDisciplines`, this.httpOptionsService.options)
+    this.http.get(`${apiUrl}/Disciplines/GetAllDisciplinesBySchool`, this.httpOptionsService.options)
```
Comment (DmMov) [inline src/app/pages/pupil-journal/pupil-journal.component.ts:31]: If you added BySchool it is not 'All' anymore. The same with back-end.

File: src/app/components/add-schedule-form/add-schedule-form.component.ts [hunk 10]
```diff
@@ -69,7 +71,7 @@ export class AddScheduleFormComponent implements OnInit {
         }
       );
 
-    this.http.get(`${apiUrl}/Disciplines/GetAllDisciplines`, this.httpOptionsService.options)
+    this.http.get(`${apiUrl}/Disciplines/GetAllDisciplinesBySchool`, this.httpOptionsService.options)
```
Comment (DmMov) [inline src/app/components/add-schedule-form/add-schedule-form.component.ts:74]: As I said before it's not 'All'.

File: src/app/components/children-info/children-info.component.ts [hunk 11]
```diff
@@ -38,13 +57,33 @@ export class ChildrenInfoComponent implements OnInit {
...
 
   ngOnInit(): void {
     this.getPupilsDataByParent();
     this.httpOptionsService.loadHeaders();
+
+    this.form = this.formBuilder.group({
+      "disciplineId": [null, requiredValidator("disciplines is required")]
+    })
+
+    this.http.get(`${apiUrl}/Disciplines/GetAllDisciplinesBySchool`, this.httpOptionsService.options)
```
Comment (DmMov) [inline src/app/components/children-info/children-info.component.ts:69]: Don't forget about it.

**Review by DmMov — APPROVED** [review]

**Review by Karonuk — APPROVED** [review]

**Review by Hrabarchuk — APPROVED** [review]

**Pull request merged**